code/modules/multiz/mobile_ladder.dm

```diff
@@ -0,0 +1,104 @@
+/obj/item/mobile_ladder
+	name = "mobile ladder"
+	desc = "A lightweight deployable ladder, which you can use to move up or down. Or alternatively, you can bash some faces in."
+	icon = 'icons/obj/multiz_items.dmi'
+	icon_state = "mobile_ladder"
+	throw_range = 3
+	force = 10
+	w_class = ITEM_SIZE_LARGE
+	slot_flags = SLOT_BACK
+
+/obj/item/mobile_ladder/proc/place_ladder(atom/A, mob/user)
+	if(isopenspace(A)) //Place into open space
+		var/turf/below_loc = GetBelow(A)
+		if (!below_loc || (istype(/turf/space, below_loc)))
+			to_chat(user, SPAN_NOTICE("Why would you do that?! There is only infinite space there..."))
+			return
+		user.visible_message(SPAN_WARNING("[user] begins to lower \the [src] into \the [A]."),
+			SPAN_WARNING("You begin to lower \the [src] into \the [A]."))
+		if (!handle_action(A, user))
+			return
+
+		var/obj/structure/ladder/mobile/downer  = new(below_loc)
+		var/obj/structure/ladder/mobile/current = new(A)
+		downer.target_up = current
+		current.target_down = downer
+
+		transfer_fingerprints_to(downer)
+		transfer_fingerprints_to(curent)
+		user.drop_from_inventory(src,get_turf(src))
+		qdel(src)
+
+	else if (istype(A, /turf/simulated/floor) || istype(A, /turf/unsimulated/floor))	//Place onto Floor
+		var/turf/upper_loc = GetAbove(A)
+		if (!upper_loc || !isopenspace(upper_loc))
+			to_chat(user, SPAN_NOTICE("There is something above. You can't deploy!"))
+			return
+		user.visible_message(SPAN_WARNING("[user] begins deploying \the [src] on \the [A]."),
+			SPAN_WARNING("You begin to deploy \the [src] on \the [A]."))
+		if (!handle_action(A, user))
+			return
+
+		var/obj/structure/ladder/mobile/current = new(A)
+		var/obj/structure/ladder/mobile/upper   = new(upper_loc)
+		current.target_up = upper
+		upper.target_down = current
+
+		transfer_fingerprints_to(current)
+		transfer_fingerprints_to(upper)
+		user.drop_from_inventory(src,get_turf(src))
+		qdel(src)
+
+/obj/item/mobile_ladder/afterattack(atom/A, mob/user,proximity)
+	if (!proximity)
+		return
+
+	place_ladder(A,user)
+
+/obj/item/mobile_ladder/proc/handle_action(atom/A, mob/user)
+	if (!do_after(user, 30, user))
+		to_chat(user, SPAN_WARNING("Can't place ladder! You were interrupted!"))
+		return FALSE
+	if (!A || QDELETED(src) || QDELETED(user))
+		// Shit was deleted during delay, call is no longer valid.
+		return FALSE
+	return TRUE
+
+/obj/structure/ladder/mobile
+	base_icon = "mobile_ladder"
+	climb_time = 4 SECONDS
+	draw_shadow = FALSE
+	tool_interaction_flags = 0
+
+/obj/structure/ladder/mobile/verb/fold()
+	set name = "Fold Ladder"
+	set category = "Object"
+	set src in oview(1)
+
+	if(usr.incapacitated() || !usr.check_dexterity(DEXTERITY_GRIP, silent = TRUE) || !ishuman(usr))
+		to_chat(usr, SPAN_WARNING("You can't do that right now!"))
```

Comment:
Dexterity should really be separate as it prints it own specific message that is more informative.